.github/workflows/quality.yml

```diff
@@ -0,0 +1,61 @@
+name: Java Code Quality Analysis
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
```

Comment:
You should ideally limit this to only running for PRs where there are actual changes in the Java code. Otherwise, this action will for every PR, even when it only affects non-Java files.